android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java

```diff
@@ -145,6 +145,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         searchView.setQuery(searchText, false);
         return true;
     }
+    @Override
+    public void onBackPressed(){
+        if((mSessions.isEmpty())&&(sessionRecyclerView.getVisibility()==View.GONE)){
```

Comment:
could you suggest a workaround for it in that case. I can't really think of anything else. Also, we might want to change the existing code and remove such conditions in that case?